Assets/Scripts/Actor/EnemyAttackManager.cs

```diff
@@ -35,100 +45,128 @@ public class EnemyAttackManager: MonoBehaviour
-                        {
-                            attackIndex = 0;
-                        }
-                        
-                        // If the enemy should die at the end of its attack cycle, kill it.
-                        if (canBeTimedOut)
-                        {
-                            vitality.TakeDamage(vitality.Health);
-                            return;
-                        }
-                        
-                        // The enemy has no more attacks if we've reached the end of the list and shouldn't loop or die.
-                        hasAttacksToExecute = false;
+                        vitality.health = 0;
+                        vitality.Die();
+                        // Disable healthbar items after last attack concludes
+                        healthBar.gameObject.SetActive(false);
+                        timeRemaining.gameObject.SetActive(false);
+                        attackNameText.gameObject.SetActive(false);
                         return;
                     }
 
-                    // Prepare for the next attack.
-                    currentAttack = attacks[attackIndex].GetComponent<EnemyAttack>();
-                    baselineHealth = vitality.Health;
-                    damageTaken = 0;
-                    timeElapsed = 0f;
-                    inAttack = false;
+                    // The enemy has no more attacks if we've reached the end of the list and shouldn't loop or die.
+                    hasAttacksToExecute = false;
+                    // Disable healthbar items after last attack concludes
+                    healthBar.gameObject.SetActive(false);
+                    timeRemaining.gameObject.SetActive(false);
+                    attackNameText.gameObject.SetActive(false);
                     return;
                 }
 
-                // Update the player's progress.
-                timeElapsed += Time.deltaTime;
-                damageTaken = baselineHealth - vitality.Health;
+                // Prepare for the next attack.
+                currentAttack = attacks[attackIndex].GetComponent<EnemyAttack>();
+                baselineHealth = vitality.Health;
+                damageTaken = 0;
+                timeElapsed = 0f;
+                inAttack = false;
+                // Disable healthbar items between attacks
+                healthBar.gameObject.SetActive(false);
+                timeRemaining.gameObject.SetActive(false);
+                attackNameText.gameObject.SetActive(false);
+                return;
             }
+
+            // Update the player's progress.
+            timeElapsed += Time.deltaTime;
+            timeRemaining.text = $"{(int)(currentAttack.durationSeconds - timeElapsed)}";
+            damageTaken = baselineHealth - vitality.Health;
+            healthBar.fillAmount = (currentAttack.health - damageTaken)/(float)currentAttack.health;
+        }
+
+        private void LateUpdate()
```

Comment:
I originally was doing some stuff here, but then I moved it out, so we can probably get rid of this method here.

Reply:
Thanks, refactored.